assets/src/edit-story/app/font/fontProvider.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Context from './context';
+
+import useLoadFonts from './effects/useLoadFonts';
+import useLoadFontFiles from './actions/useLoadFontFiles';
+
+function FontProvider( { children } ) {
+	const [ fonts, setFonts ] = useState( [] );
+
+	useLoadFonts( { fonts, setFonts } );
+
+	const getFontByName = useCallback(
```

Comment:
Would a single function something like `getFontBy` make sense here instead of two separate?
E.g. `fonts.find( ( thisFont ) => thisFont[ key ] === value );`?

(Not a strong preference if it makes code much more readable as it is, feel free to ignore).

Reply:
Not against this as removed repeated code. `getFontBy` could then just become an alias of `getFontByName` and getFontBySlug`.

Reply:
Changed in https://github.com/ampproject/amp-wp/pull/3998/commits/eac50b04f7e3d1829eaefe62772008b5f56ad8d2